### PR TITLE
AddObserver adds a node to a suite

### PR DESCRIPTION
**Pull request by m4ksio**
Makes Observer nodes part of test suite. 
- Easier cleanup with entire suite
- Output of the observer container is grabbed along with other logs, as for any other container. Previously it was ignored

### CODE REVIEW

**Review by peterargue — APPROVED** [review]
thanks for cleaning this up!

File: integration/testnet/network.go [hunk 1]
```diff
@@ -818,10 +817,17 @@ func (net *FlowNetwork) AddObserver(t *testing.T, ctx context.Context, conf *Obs
 	// Setup directories
 	tmpdir, _ := os.MkdirTemp(TmpRoot, "flow-integration-node")
```
Comment (SaveTheRbtz) [inline integration/testnet/network.go:818]: You should probably use [t.TempDir](https://pkg.go.dev/testing#T.TempDir) here

File: integration/testnet/network.go [hunk 2]
```diff
@@ -818,10 +817,17 @@ func (net *FlowNetwork) AddObserver(t *testing.T, ctx context.Context, conf *Obs
...
 	flowDataDir := filepath.Join(tmpdir, DefaultFlowDataDir)
+	err = os.Mkdir(flowDataDir, 0700)
+	require.NoError(t, err)
 	nodeBootstrapDir := filepath.Join(tmpdir, DefaultBootstrapDir)
+	err = os.Mkdir(nodeBootstrapDir, 0700)
+	require.NoError(t, err)
 	flowProfilerDir := filepath.Join(flowDataDir, "./profiler")
+	err = os.Mkdir(flowProfilerDir, 0700)
+	require.NoError(t, err)
```
Comment (SaveTheRbtz) [inline integration/testnet/network.go:827]: a helper function maybe?

File: integration/testnet/network.go [hunk 3]
```diff
@@ -831,71 +837,72 @@ func (net *FlowNetwork) AddObserver(t *testing.T, ctx context.Context, conf *Obs
...
+		HealthCheck: testingdock.HealthCheckCustom(healthcheckAccessGRPC(observerUnsecurePort)),
 	}
 
-	err = net.cli.ContainerStart(ctx, container.ID, types.ContainerStartOptions{})
-	if err != nil {
-		return nil, err
+	suiteContainer := net.suite.Container(containerOpts)
+
+	nodeContainer := &Container{
+		//Config:  nodeConf,
```
Comment (SaveTheRbtz) [inline integration/testnet/network.go:893]: remove?

File: integration/tests/access/observer_test.go [hunk 4]
```diff
@@ -93,13 +89,7 @@ func (suite *ObserverSuite) SetupTest() {
 	})
 	require.NoError(suite.T(), err)
 
-	time.Sleep(time.Second * 3) // needs breathing room for the observer to start listening
```
Comment (SaveTheRbtz) [inline integration/tests/access/observer_test.go:96]: thank you for adding proper healthchecks.

**Review by SaveTheRbtz — APPROVED** [review]

**Review by SaveTheRbtz — APPROVED** [review]
Thank you for cleaning it up!

**Review by koko1123 — APPROVED** [review]

**Pull request merged**